common/utils/query_utils.ts

```diff
@@ -23,14 +23,84 @@ const escapeQuotes = (literal: string) => {
   return literal.replaceAll("'", "''");
 };
 
-export const getIndexPatternFromRawQuery = (query: string): string => {
+const PROMQL_INDEX_REGEX = /(?<connection>[^|\s]+)\.query_range\('(?<promql>.+?)'/i;
+const PROMQL_AGG_ATTR_REGEX = /(?<agg>\S+) *?(by\((?<attrs>.*?)\))? *?\((?<metric>\S+)\)/i;
+const PROMQL_METRIC_ONLY_REGEX = /^[(^\s]*?(?<metric>[_.a-z0-9]+)\)?/i;
+
+const promQlMetricOnly = (promQL: string) => promQL.match(PROMQL_METRIC_ONLY_REGEX)?.groups;
+const promQLAggAttr = (promQL: string) => promQL.match(PROMQL_AGG_ATTR_REGEX)?.groups;
+
+interface PromQLAST {
+  connection: string;
+  metric: string;
+  agg?: string;
+  attrs?: string;
+}
+
+const promQLMatch = (query: string): PromQLAST | undefined => {
```

Comment:
this function seems to be extracting keywords from string query, why is it called AST?

Reply:
Renamed throughout.